lms/djangoapps/course_api/blocks/transformers/tests/test_milestones.py

```diff
@@ -0,0 +1,280 @@
+    update_attempt_status
+)
+from edx_proctoring.models import ProctoredExamStudentAttemptStatus
+from edx_proctoring.runtime import set_runtime_service
+from edx_proctoring.tests.test_services import MockCreditService
+from gating import api as lms_gating_api
+from lms.djangoapps.course_blocks.transformers.tests.helpers import CourseStructureTestCase
+from opaque_keys.edx.keys import UsageKey
+from student.tests.factories import CourseEnrollmentFactory
+
+from ..milestones import MilestonesTransformer
+from ...api import get_course_blocks
+from openedx.core.lib.gating import api as gating_api
+from milestones.tests.utils import MilestonesTestCaseMixin
+
+
+@attr('shard_3')
+@ddt.ddt
+@patch.dict('django.conf.settings.FEATURES', {'ENABLE_SPECIAL_EXAMS': True, 'MILESTONES_APP': True})
+class MilestonesTransformerTestCase(CourseStructureTestCase, MilestonesTestCaseMixin):
+    """
+    Test behavior of ProctoredExamTransformer
+    """
+    TRANSFORMER_CLASS_TO_TEST = MilestonesTransformer
+
+    def setUp(self):
+        """
+        Setup course structure and create user for split test transformer test.
+        """
+        super(MilestonesTransformerTestCase, self).setUp()
+
+        # Set up proctored exam
+
+        # Build course.
+        self.course_hierarchy = self.get_course_hierarchy()
+        self.blocks = self.build_course(self.course_hierarchy)
+        self.course = self.blocks['course']
+
+        # Enroll user in course.
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id, is_active=True)
+
+    def setup_proctored_exam(self, block, attempt_status, user_id):
+        """
+        Test helper to configure the given block as a proctored exam.
+        """
+        exam_id = create_exam(
+            course_id=unicode(block.location.course_key),
+            content_id=unicode(block.location),
+            exam_name='foo',
+            time_limit_mins=10,
+            is_proctored=True,
+            is_practice_exam=block.is_practice_exam,
+        )
+
+        set_runtime_service(
+            'credit',
+            MockCreditService(enrollment_mode='verified')
+        )
+
+        create_exam_attempt(exam_id, user_id, taking_as_proctored=True)
+        update_attempt_status(exam_id, user_id, attempt_status)
+
+    def setup_gated_section(self, gated_block, gating_block):
+        """
+        Test helper to create a gating requirement.
+        Args:
+            gated_block: The block that should be inaccessible until gating_block is completed
+            gating_block: The block that must be completed before access is granted
+        """
+        gating_api.add_prerequisite(self.course.id, unicode(gating_block.location))
+        gating_api.set_required_content(self.course.id, gated_block.location, gating_block.location, 100)
+
+    ALL_BLOCKS = ('course', 'A', 'B', 'C', 'ProctoredExam', 'D', 'E', 'PracticeExam', 'F', 'G', 'H', 'I',
+                  'TimedExam', 'J', 'K')
```

Comment:
Nit: formatting convention - have opening and closing parens on their own lines.  So it would look like:

```
ALL_BLOCKS = (
    'course', 'A', 'B', 'C', 'ProctoredExam', 'D', 'E', 'PracticeExam', 'F', 'G', 'H', 'I', 'TimedExam', 'J', 'K'
)
```